xinference/model/llm/utils.py

```diff
@@ -670,35 +670,6 @@ def _eval_tool_arguments(cls, model_family, c, tools):
         logger.debug("Tool call content: %s, func: %s, args: %s", content, func, args)
         return content, func, args
 
-    @classmethod
-    def _tools_token_filter(cls, model_family):
-        """
-        Generates a filter function for Qwen series models to retain outputs after "\nFinal Answer:".
-
-        Returns:
-            A function that takes tokens (string output by the model so far) as input
-            returns True if current token is after "\nFinal Answer:", else False.
-        """
-        family = model_family.model_family or model_family.model_name
-        if family in ["qwen-chat", "qwen1.5-chat"]:
-            # Encapsulating function to reset 'found' after each call
-            found = False
-
-            def process_token(tokens: str):
-                nonlocal found
-                # Once "Final Answer:" is found, future tokens are allowed.
-                if found:
-                    return True
-                # Check if the token ends with "\nFinal Answer:" and update `found`.
-                if tokens.endswith("\nFinal Answer:"):
```

Comment:
This part similar to the official `openai_api.py`: https://github.com/QwenLM/Qwen/blob/main/openai_api.py#L346 Are you sure it's OK to remove it?

Reply:
> This part is similar to the official `openai_api.py`: https://github.com/QwenLM/Qwen/blob/main/openai_api.py#L346 Are you sure it's OK to remove it?

Can Qwen offical `openai_api.py` support tools with stream response? I'm not sure.

Reply:
From the qwen official `openai_api.py`: https://github.com/QwenLM/Qwen/blob/main/openai_api.py#L410 It says that function calling does not work in stream mode.

Reply:
![image](https://github.com/xorbitsai/inference/assets/138426806/c8359845-4505-4137-88dc-a01c47e0cc50)

Reply:
Is it possible that the `\nFinal Answer:` is not in one token?

Reply:
For example, `Final` is token a,  `Answer` is token b. If this is True, we can use a two-tokens buffer to check the  `Final Answer` string.

Reply:
After debugging, I realized that the reason for the error was most likely that the model was outputting `Final answer` instead of `Final Answer`.

Reply:
vllm's LLMEngine doesn't return tokens one by one, it may return a bunch of tokens at a time

Reply:
I committed the new implementation, which should fix the bug and keep the filter. @codingl2k1

Reply:
Nice work! Could you add a test for it?